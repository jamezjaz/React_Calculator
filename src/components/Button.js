import React from 'react';
import PropTypes from 'prop-types';

const Button = props => {
  const { name, handleClick } = props;
  return (
    <div>
      <p>
        <button
          type="button"
          className="cal-button"
          onClick={handleClick}
        >
          {name}
        </button>
      </p>
    </div>
  );
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Button;
