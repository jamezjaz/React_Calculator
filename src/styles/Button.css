button {
  font-size: 40px;
  width: 180px;
  height: 100px;
  border: 1.5px solid #d1d1d1;
  outline: none;
}

button:hover {
  background-color: #bebbbb;
}

.orange {
  background-color: #f3903e;
}

.orange:hover {
  background-color: #ce7934;
}

.wide {
  width: 360px;
}
