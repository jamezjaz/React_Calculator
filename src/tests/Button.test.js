import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import Button from '../components/Button';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });

describe('<Button />', () => {

  it('renders correctly', () => {
    const tree = renderer
      .create(<Button />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it contains a button', () => {
    let wrapper = shallow(<Button />);
    const button = <button></button>;
    expect(wrapper.containsMatchingElement(button)).toBe(true);
  });

  it('renders the button texts correctly', () => {
    const { getByText } = render(<Button name="รท" />);
    const divButton = getByText(/รท/i);
    expect(divButton).toBeInTheDocument();
  });

  it('the buttons play the role of buttons', () => {
    const { getByRole } = render(<Button name="รท" />);
    const divButton = getByRole('button');
    expect(divButton).toBeInTheDocument();
  });
});