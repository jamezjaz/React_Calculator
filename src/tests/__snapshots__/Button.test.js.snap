// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<div>
  <p>
    <button
      className="orange"
      type="button"
    />
  </p>
</div>
`;
